app/models/future_role.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+#
+# Copyright (c) 2023, Schweizer Alpen-Club. This file is part of
+# hitobito_sac_cas and licensed under the Affero General Public License version 3
+# or later. See the COPYING file at the top-level directory or at
+# https://github.com/hitobito/hitobito
+
+class FutureRole < Role
+  self.kind = :future
+  self.basic_permissions_only = true
+
+  IGNORED_ATTRS = %w(id type convert_on convert_to created_at).freeze
+
+  skip_callback :create, :after, :set_first_primary_group
+  skip_callback :create, :after, :set_contact_data_visible
+  skip_callback :destroy, :after, :reset_contact_data_visible
+  skip_callback :destroy, :after, :reset_primary_group
+
+  validates :person, :group, presence: true
+  validates :convert_to, inclusion: { within: :group_role_types }, if: :group
```

Comment:
In welchem Fall hat eine `FutureRole` keine `group`?

Reply:
Wahrscheinlich immer, aber ein `Role.new.valid?` würde einen Fehler produzieren, da keine Gruppe verfügbar ist um die `inclusion` validierung zu machen.